aptos-move/move-examples/rewards_pool/sources/rewards_pool.move

```diff
@@ -0,0 +1,211 @@
+/// An example module that manages rewards for multiple tokens on a per-epoch basis. Rewards can be added for multiple
+/// tokens by anyone for any epoch but only friended modules can increase/decrease claimer shares.
+///
+/// This module is designed to be integrated into a complete system that manages epochs and rewards.
+///
+/// The flow works as below:
+/// 1. A rewards pool is created with a set of reward tokens (fungible assets). If coins are to be used as rewards,
+/// developers can use the coin_wrapper module from move-examples/swap to convert coins into fungible assets.
+/// 2. Anyone can add rewards to the pool for any epoch for multiple tokens by calling add_rewards.
+/// 3. Friended modules can increase/decrease claimer shares for current epoch by calling increase_allocation and
+/// decrease_allocation.
+/// 4. Claimers can claim their rewards for any epoch that has ended by calling claim_rewards.
+///
+/// Note that due to integer division, there can be rounding errors that leave 1 unit of reward left for each token in
```

Comment:
how does this happen, because you don't update the pool?

Reply:
Yes because no redemption per a comment below. Im experimenting with only allowing all tokens to be claimed at once so we can clean up the pool

Reply:
I think now I get it.. a little bit overcomplicated...

Reply:
How would you have done it more simply?

Reply:
No, I didn't mean it can be done simply if you just have one pool and support withdraw separately. As you said, withdraw all at once... I mean, withdraw separately may be over complicated and not that important to have

Reply:
Updated code to be simpler and more consistent